src/accordion/accordion.ts

```diff
@@ -32,6 +32,36 @@ export class NgbPanelContent {
 }
 
 /**
+ * This directive must be used to wrap accordion panel header.
+ */
+@Directive({selector: 'template[ngbPanelHeader]'})
+export class NgbPanelHeader {
+  constructor(public templateRef: TemplateRef<any>) {}
+}
+
+/**
+ * The context available to the NgbPanelHeader directive
+ */
+export class NgbPanelHeaderContext implements NgbPanelHeaderContext {
```

Comment:
Class and interface with the same name? We should distinguish those to avoid confusion. `NgbPanelHeaderContextImpl` for the implementation class?